### PR TITLE
Add write_more to support mjpeg streaming

### PR DESCRIPTION
**Pull request by dsmyth**
This was based on code found at https://github.com/aanon4/RobotVision -- they just pulled in Simple-Web-Server and modified it locally rather than pushing it back as a pull request.

MJPEG resource:
    // mjpeg stream resource
    server.resource["^/stream"]["GET"] = [this](std::shared_ptr<SimpleWeb::Server<SimpleWeb::HTTP>::Response> response,
                                                std::shared_ptr<SimpleWeb::Server<SimpleWeb::HTTP>::Request> request) {
        response->set_nodelay();
        std::thread streamThread([this, response] {
            class MJPEG {
            public:
                static void send_image(RestServer* server, const std::shared_ptr<SimpleWeb::Server<SimpleWeb::HTTP>::Response> &response) {
                    std::unique_lock<std::mutex> lk(server->streamMutex);
                    server->streamCv.wait(lk);
                    int size = (int)server->vJPEG.size();
                    response->write_more("--mjpegstream\r\n");
                    response->write_more("Content-Type: image/jpeg\r\n");
                    response->write_more(str(boost::format("Content-Length: %1%\r\n\r\n") % size));
                    response->write((const char*)server->vJPEG.data(), size);
                    lk.unlock();
                    response->write_more("\r\n");
                    response->send([server, response](const SimpleWeb::error_code &ec) {
                        if (!ec) {
                            send_image(server, response);
                        }
                        else {
                            // Done
                            if (server->active > 0) {
                                server->active--;
                            }
                        }
                    });
                }
            };
            SimpleWeb::CaseInsensitiveMultimap headers;
            headers.emplace("Content-Type", "multipart/x-mixed-replace;boundary=mjpegstream");
            headers.emplace("Connection", "close");
            headers.emplace("Cache-Control", "no-cache, no-store, max-age=0, must-revalidate");
            headers.emplace("Content-Length", "-1");
            response->write(SimpleWeb::StatusCode::success_ok, headers);
            active++;
            MJPEG::send_image(this, response);
        });
        streamThread.detach();
    };

and image update for openCV:
class MJPEGServer {
    SimpleWeb::Server<SimpleWeb::HTTP> server;
    std::mutex streamMutex;
    std::condition_variable streamCv;
    std::vector<uchar> vJPEG;
    unsigned int active = 0;

void setStreamJPEG(cv::Mat &streamMat) {
    // only do the work if we are actively streaming
    if (active > 0 && streamMutex.try_lock())
    {
        cv::imencode(".jpg", streamMat, vJPEG, std::vector<int>());
        streamCv.notify_all();
        streamMutex.unlock();
    }
}
}

### CODE REVIEW

File: server_http.hpp [hunk 1]
```diff
@@ -146,6 +146,15 @@ namespace SimpleWeb {
         write(StatusCode::success_ok, std::string(), header);
       }
 
+      void write_more(const std::string &content) {
+        *this << content;
+      }
+
+      void set_nodelay() {
+        asio::ip::tcp::no_delay option(true);
+        session->connection->socket->set_option(option);
```
Comment (eidheim) [inline server_http.hpp]: Thank you, but the no_delay option is already set here: https://github.com/eidheim/Simple-Web-Server/blob/master/server_http.hpp#L712

File: server_http.hpp [hunk 2]
```diff
@@ -146,6 +146,15 @@ namespace SimpleWeb {
         write(StatusCode::success_ok, std::string(), header);
       }
 
+      void write_more(const std::string &content) {
+        *this << content;
+      }
```
Comment (eidheim) [inline server_http.hpp]: The write methods are just convenient methods for creating responses. For other cases, using the response stream through the `<<` operator is probably better than having a `write_more(const std::string&)` function, since the `<<` operator works with several other parameters as well.

File: server_http.hpp (hunk 1 above)
Comment (dsmyth) [reply server_http.hpp]: Ah! I missed that.

File: server_http.hpp (hunk 2 above)
Comment (dsmyth) [reply server_http.hpp]: I'll give that a try.

Comment (dsmyth) [reply server_http.hpp]: That works just fine. I've deleted my fork & gone back to your master branch. My new response code for mjpeg streaming now looks like this -- feel free to use as a streaming example if you like:

    // mjpeg stream resource
    server.resource["^/stream"]["GET"] = [this](std::shared_ptr<SimpleWeb::Server<SimpleWeb::HTTP>::Response> response,
                                                std::shared_ptr<SimpleWeb::Server<SimpleWeb::HTTP>::Request> request) {
        BOOST_LOG_TRIVIAL(info) << "GET /stream";
        std::thread streamThread([this, response] {
            class MJPEG {
            public:
                static void send_image(RestServer* server, const std::shared_ptr<SimpleWeb::Server<SimpleWeb::HTTP>::Response> &response) {
                    std::unique_lock<std::mutex> lk(server->streamMutex);
                    server->streamCv.wait(lk);
                    int size = (int)server->vJPEG.size();
                    *response << "--mjpegstream\r\n"
                            << "Content-Type: image/jpeg\r\n"
                            << str(boost::format("Content-Length: %1%\r\n\r\n") % size);
                    response->write((const char*)server->vJPEG.data(), size);
                    lk.unlock();
                    *response << "\r\n";
                    response->send([server, response](const SimpleWeb::error_code &ec) {
                        if (!ec) {
                            send_image(server, response);
                        }
                        else {
                            // Done
                            if (server->active > 0) {
                                server->active--;
                            }
                            BOOST_LOG_TRIVIAL(info) << "GET /image active--: " << server->active;
                        }
                    });
                }
            };
            SimpleWeb::CaseInsensitiveMultimap headers;
            headers.emplace("Content-Type", "multipart/x-mixed-replace;boundary=mjpegstream");
            headers.emplace("Connection", "close");
            headers.emplace("Cache-Control", "no-cache, no-store, max-age=0, must-revalidate");
            headers.emplace("Content-Length", "-1");
            response->write(SimpleWeb::StatusCode::success_ok, headers);
            active++;
            BOOST_LOG_TRIVIAL(info) << "GET /image active++: " << active;
            MJPEG::send_image(this, response);
        });
        streamThread.detach();
    };

Comment (eidheim) [reply server_http.hpp]: Thank you, the example is much appreciated!
